Sources/HummingbirdWebSocket/WebSocketChannel.swift

```diff
@@ -254,16 +254,16 @@ public struct HTTP1WebSocketUpgradeChannel: ServerChildChannel, HTTPChannelHandl
         }
     }
 
-    /// Upgrade failed we should write a 405
-    private func write405(asyncChannel: NIOAsyncChannel<HTTPRequestPart, HTTPResponsePart>, logger: Logger) async {
+    /// Write empty HTTP response for failed upgrade
+    private func writeFailedUpgrade(asyncChannel: NIOAsyncChannel<HTTPRequestPart, HTTPResponsePart>, logger: Logger) async {
         do {
             try await asyncChannel.executeThenClose { _, outbound in
                 let headers: HTTPFields = [
                     .connection: "close",
                     .contentLength: "0",
                 ]
                 let head = HTTPResponse(
-                    status: .methodNotAllowed,
+                    status: .ok,
```

Comment:
Isn't this more of a `500` / `internalServerError`?

Reply:
It's not a 5xx as it isn't a server error. It is caused by the canUpdate failing. But the standard for an upgrading failing is 200. A successful upgrade returns a 101.